### DISCUSSION TITLE
FR: Command Palette

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#820] [Ideas]
### What feature would you like to see?

Suggestion: Improve accessibility of functions without relying on keyboard shortcuts

I find it challenging to memorize numerous keyboard shortcuts. It would be beneficial to have the ability to access all functions, such as "Split View Vertical", through a command palette-like interface similar to what VSCode offers.

This feature would:
1. Reduce the learning curve for new users
2. Improve accessibility for users who prefer not to use keyboard shortcuts

Could we consider implementing a searchable command palette to make all functions easily accessible without the need for memorizing keyboard shortcuts?

### What platform are you seeing the problem on?

Windows

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/494_

**Comment by billyjacoby** [comment]
Just adding my 2 cents here, this is one feature that would certainly make it easier for me to make the full time move from Arc to Zen Browser

**Reply by yudjinn** [reply]
It would also help me move from Vivaldi to Zen. I frequently open the command palette to paste links to navigate to quickly

**Comment by wsippel** [comment]
Love the idea. However, instead of a separate command palette, I believe it should simply be integrated with the URL bar, with a leader to switch between typing a URL and a command. Maybe `/` or `#`, something URLs don't begin with. I'd also suggest using the standard CTRL+Shift+P to bring up the URL entry/ command palette, move 'new private tab' to CTRL+Shift+N (like Chromium based browsers), and make the process manager available via a command from the palette. Dev tools and other advanced stuff are perfect candidates for a command palette. Also obviously toggling fullscreen mode, anything related to tiling, pinning and workspace management.

**Reply by borghiandrey** [reply]
you are literally me

**Comment by snowman** [comment]
Especially useful when install lots of extensions and to open extension menu without needing eyeball search the extension icon to click and without showing installed extensions in the toolbar to make layout clean

see [Vivaldi Quick Commands](https://help.vivaldi.com/desktop/shortcuts/quick-commands/)

**Comment by michaelthatsit** [comment]
Just switching over from Arc. this is probably the one thing I'm missing most.

**Reply by vraravam** [reply]
Me as well. I have almost uninstalled Arc - except for the command palette functionality!

**Reply by yashcrest** [reply]
@michaelthatsit @vraravam where can I find command palette option on arc ? Are you referring to (cmd + T ) search bar?

**Comment by BenjaminJacober** [comment]
I like the way Arc Browser does this. Opening a new tab and changing the current one (Ctrl + L) open the same search interface. There you can search for settings (in general or a specific setting), extensions and browser functions. It works like an IDE general search, (IntelliJ, Fleet are the only ones I've ever used).

**Comment by unkindypie** [comment]
will this feature be implemented?

**Comment by mikeckennedy** [comment]
I'm here to add my 2 cents. Some sort of command palette with cmd+k or cmd+e like Vivaldi and other browsers would be very much appreciated.

**Comment by Ulset** [comment]
Bump to get the command palette from Arc (CTRL + T and CTRL + L)

**Comment by ayyjohn** [comment]
would love to see vivaldi's take on this be implemented exactly!

**Comment by DianP** [comment]
👍 +1

The reason why I'm still on my arc both on Mac & Windows is because of the command palette. It's easier to operate because I don't have to remember every shortcut. 

Aside from the functionality, I would love to see the same look as Arc, where the command palette stays in the centre of the screen.

If Zen implemented this, it would be amazing. at least just for the new tab ( CTRL + T ) function and the (CTRL + L ) URL bar function is good for the starting point.

**Comment by EmirOven** [comment]
Bump

This is the one thing I am missing coming over Vivaldi, especially as a programmer using VSCode this would be such a productivity-increaser!

**Comment by unkindypie** [comment]
Bump

**Comment by mr-cheffy** [comment]
Guys, no need to bump it. I would like to focus on stability before adding some new stuff if that's ok with y'all. Adding too many features without focusing on bug fixing can be damanging

**Reply by italoaa** [reply]
Responsable dev. you are going far.

**Comment by devployment** [comment]
Just for my curiosity. If someone might want to contribute to this, what would be a good place to start? 

Feels like it might be a bit more involved, as this feature is not available in "core-FF". I can imagine that offering a "command palette" to existing FF actions might be somehow done by listing and triggering available actions in a "custom UI". But it might be a bit more involved to integrate add on actions. e.g. opening them from that command bar.

Anyone with pointers where to start looking at?

**Reply by 2zqa** [reply]
I think it could, or actually should, be integrated in the URL bar. Similar to VS Code, there's one search bar and depending on the prefix (">", "#", "@", etc) it finds different actions. Firefox already supports a few: %, \^ and \*.

Apparently, Firefox even already supports a few actions like "refresh Firefox". Read more about it here: https://support.mozilla.org/kb/search-firefox-address-bar

**Comment by ChristinWhite** [comment]
Where I'm really, really missing this functionality is manipulating workspaces. There are a number of command palette extensions that cover a lot of the basic browsing functions, and I'm pretty happy with Surfingkeys for those functions. The biggest issue is the inability to navigate between workspaces and, especially, to move tabs to other workspaces by name. Right now, almost the only time I have to use a mouse doing workspace-related interaction.

Just adding workspace and other Zen specific feature APIs that we could utilize with existing extensions would be a huge improvement. In addition to workspaces, I think the ability to manipulate tiling, select multiple tabs and toggle compact mode would cover most of the bases.

**Reply by yashcrest** [reply]
I have mapped the shortcut (cmd + option + leftArrow / rightArrow) to navigate between different workspaces for my settings. Is this what you were after ? Or wanted a SurfingKeys key binding for switching between workspaces? 

If yes, I find this feature also missing in Arc browser as well.

**Reply by ChristinWhite** [reply]
I have something similar setup now, but when you have 22 different workspaces you really need to be able to type a name instead of manually navigating up and down the workspaces. Ideally, I'd be able to enter something into a command palette like `News` (or `Go to News`) and `Change Tabs To Workspace News`. Vivaldi has this, I think Arc and SigmaOS do too.

I can get away with not having a go-to workspace function if I remember a tab in it, it's moving that is more challenging.

A simpler alternative would be to just expose the existing `Change Tab(s) To Workspace / News` command in the right click menu to the app menus and maybe add a `Go To Workspace / News` menu to the application menus so that a we could use UI scripting and a third party command palette like Alfred, Paletro, Raycast or Keyboard Maestro to fire those commands. I'm assuming similar applications exist on Linux and Windows (AutoHotkey maybe).

**Comment by wundersmiths** [comment]
Native FF has 'quick actions' which have some of the more useful commands handy - I wonder why that is not available?
![image](https://github.com/user-attachments/assets/62955841-2003-4b15-be66-fdd1634cc692)

**Comment by devployment** [comment]
For everyone that might have missed it like I did. Twilight now has some command palette going on. Very excited to see this happening. 🫶🏼

https://zen-browser.app/release-notes/

The last thing that I'm missing is native rocker gestures. When that arrives I can make Zen finally my default browser. :)

**Reply by vraravam** [reply]
What's the key combination to trigger this?

**Reply by devployment** [reply]
It's just the search bar that is opened with cmd+t (macos). It contains additional "commands" that can be executed. 

Defined here

https://github.com/zen-browser/desktop/blob/23bdc994d41bcb6cf12e74b0124403a20d37f0ac/src/zen/urlbar/ZenUBGlobalActions.sys.mjs

<img width="746" height="269" alt="image" src="https://github.com/user-attachments/assets/29e2b8d2-695a-4ea2-889f-dce442ec33ff" />

Or open extensions

<img width="1020" height="404" alt="Snagit 2025-09-20 14 47 45" src="https://github.com/user-attachments/assets/405ae939-68b6-4060-b08c-2724091b3040" />

**Reply by hajiboy95** [reply]
very weird, i justq downloaded the newest version but for me it seems not working even though it's supposed to be on 16.1b :( 
<img width="700" height="165" alt="grafik" src="https://github.com/user-attachments/assets/93b2eb35-24d3-4061-bea3-df029bafc3cf" />

**Reply by hajiboy95** [reply]
do i need to activate something within the settings?

**Reply by kvenn** [reply]
Are you on Zen Twilight? I'm seeing it
https://zen-browser.app/download/?twilight

**Reply by Ivaon** [reply]
> do i need to activate something within the settings?

@hajiboy95 The command palette has yet to be localized, so you need to set your browser in en-US to be able to use them.

**Reply by hajiboy95** [reply]
@Ivaon thank you so much, I'll give it a try !! Probably in settings, right?  

@kvenn No I'm not on twilight but in the recent release notes for the official release it's also been listed :D

**Reply by Ivaon** [reply]
@hajiboy95 It's indeed in the settings, in general, scroll down a bit, it's under language: 

<img width="831" height="159" alt="image" src="https://github.com/user-attachments/assets/35e16bca-6870-43c1-95ad-022d7ef9e291" />

**Reply by snowman** [reply]
which keyboard shortcut to call it?

Edit:
Just press `Ctrl-t`, you can now search extension, switch tabs, open settings etc with `1.17.3b` version
